### PR TITLE
Upgrade to latest MC image

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
Upgrading MC to https://github.com/kubermatic/machine-controller/commit/d62ab03c71afd83c2d13549b06f85bd4621ef186 and removing all the deprecated flags for MC deployment.

I'm intentionally not updating the go dependency since that would require bumping the k8s API, etc. Should be upgraded before the next minor release.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```
/assign
/cc @kron4eg

### CODE REVIEW

File: addons/machinecontroller/deployment-controller.yaml [hunk 1]
```diff
@@ -45,35 +45,14 @@ spec:
           args:
             - -health-probe-address=0.0.0.0:8085
             - -metrics-address=0.0.0.0:8080
-            {{ if .Config.Features.NodeLocalDNS.Deploy -}}
-            - -cluster-dns={{ .Resources.NodeLocalDNSVirtualIP }}
```
Comment (xmudrii) [inline addons/machinecontroller/deployment-controller.yaml:49]: FYI: This flag is not marked as deprecated in MC but seems to be no-op.

**Review by xmudrii — CHANGES_REQUESTED** [review]
Can you also please remove legacy MC tests from `tests.yml` (including relevant scenario and infrastructure)?

**Comment by xmudrii** [comment]
@ahmedwaleedmalik Can you also remove:
- https://github.com/kubermatic/kubeone/blob/b9768c8495faab44e8441df72ee625233a5ad968/test/e2e/tests_definitions.go#L1413-L1420
- https://github.com/kubermatic/kubeone/blob/main/test/e2e/testdata/legacy_machine_controller_containerd_external.yaml

**Comment by ahmedwaleedmalik** [comment]
> * legacy_machine_controller_containerd_external

Marko, I will, but please keep in mind that this PR was only to ensure that the flags were correct.

**Comment by ahmedwaleedmalik** [comment]
/hold
Some indentation issue in the webhook.yaml, will check back later.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**